applications/FluidDynamicsApplication/custom_conditions/stokes_wall_condition.cpp

```diff
@@ -99,58 +127,53 @@ void StokesWallCondition<3,3>::GetDofList(DofsVectorType& rElementalDofList,
 }
 
 
-
-template <>
-void StokesWallCondition<2,2>::CalculateNormal(array_1d<double,3>& An)
-{
-    Geometry<Node<3> >& pGeometry = this->GetGeometry();
-
-    An[0] =   pGeometry[1].Y() - pGeometry[0].Y();
-    An[1] = - (pGeometry[1].X() - pGeometry[0].X());
-    An[2] =    0.00;
-
-}
-
+/**
+ * @see StokesWallCondition::GetDofList
+ */
 template <>
-void StokesWallCondition<3,3>::CalculateNormal(array_1d<double,3>& An )
+void StokesWallCondition<3,4>::GetDofList(
+    DofsVectorType& rElementalDofList,
+    const ProcessInfo& rCurrentProcessInfo) const
 {
-    Geometry<Node<3> >& pGeometry = this->GetGeometry();
+    const SizeType NumNodes = 4;
+    const SizeType LocalSize = 16;
 
-    array_1d<double,3> v1,v2;
-    v1[0] = pGeometry[1].X() - pGeometry[0].X();
-    v1[1] = pGeometry[1].Y() - pGeometry[0].Y();
-    v1[2] = pGeometry[1].Z() - pGeometry[0].Z();
+    if (rElementalDofList.size() != LocalSize)
+        rElementalDofList.resize(LocalSize);
 
-    v2[0] = pGeometry[2].X() - pGeometry[0].X();
-    v2[1] = pGeometry[2].Y() - pGeometry[0].Y();
-    v2[2] = pGeometry[2].Z() - pGeometry[0].Z();
+    unsigned int LocalIndex = 0;
 
-    MathUtils<double>::CrossProduct(An,v1,v2);
-    An *= 0.5;
+    for (unsigned int iNode = 0; iNode < NumNodes; ++iNode)
+    {
+        rElementalDofList[LocalIndex++] = this->GetGeometry()[iNode].pGetDof(VELOCITY_X);
+        rElementalDofList[LocalIndex++] = this->GetGeometry()[iNode].pGetDof(VELOCITY_Y);
+        rElementalDofList[LocalIndex++] = this->GetGeometry()[iNode].pGetDof(VELOCITY_Z);
+        rElementalDofList[LocalIndex++] = this->GetGeometry()[iNode].pGetDof(PRESSURE);
+    }
 }
 
 template<unsigned int TDim, unsigned int TNumNodes>
 void StokesWallCondition<TDim,TNumNodes>::ApplyNeumannCondition(MatrixType &rLocalMatrix, VectorType &rLocalVector)
 {
     const unsigned int LocalSize = TDim+1;
     const GeometryType& rGeom = this->GetGeometry();
-    const GeometryType::IntegrationPointsArrayType& IntegrationPoints = rGeom.IntegrationPoints(GeometryData::GI_GAUSS_2);
+    const GeometryData::IntegrationMethod integration_method = static_cast<GeometryData::IntegrationMethod> (GeometryData::GI_GAUSS_2);
+    const GeometryType::IntegrationPointsArrayType& IntegrationPoints = rGeom.IntegrationPoints(integration_method);
     const unsigned int NumGauss = IntegrationPoints.size();
+    Vector gauss_pts_J_det = ZeroVector(NumGauss);
+    rGeom.DeterminantOfJacobian(gauss_pts_J_det, integration_method);
+    MatrixType NContainer = rGeom.ShapeFunctionsValues(integration_method);
 
-    MatrixType NContainer = rGeom.ShapeFunctionsValues(GeometryData::GI_GAUSS_2);
-
+    // Compute Unit Normal
     array_1d<double,3> Normal;
-    this->CalculateNormal(Normal); //this already contains the area
+    rGeom.Normal(Normal);
```

Comment:
Sorry for the buzzkill, but this call is wrong.  This would corrupt even the stokes simulation with tet mesh. I came to know this when I tried to run a simple example as a final check. 

Another thing to take note of is that a test for Stokes wall condition is missing.  Fortunately, we have one for the Navier-Stokes wall condition.  Probably it's about time to take up the cleanup and enhancement of these conditions as the next PR.

Reply:
Thanks for reporting. Let's do the upcoming changes ASAP then.